### PR TITLE
Enable VSTHRD200 (Use "Async" suffix for async methods)

### PR DESCRIPTION
**Pull request by sharwell**
This rule relies on a suppression analyzer (#4803) to avoid requiring test methods to be renamed.

### CODE REVIEW

File: src/Microsoft.ML.Sweeper/AsyncSweeper.cs [hunk 1]
```diff
@@ -47,7 +47,7 @@ public interface IAsyncSweeper
         /// Propose a <see cref="ParameterSet"/>.
         /// </summary>
         /// <returns>A future <see cref="ParameterSet"/> and its id. Null if unavailable or cancelled.</returns>
-        Task<ParameterSetWithId> Propose();
+        Task<ParameterSetWithId> ProposeAsync();
```
Comment (sharwell) [inline src/Microsoft.ML.Sweeper/AsyncSweeper.cs]: 📝 This is the only public API change in this pull request. It's not clear to me if/when this API shipped or whether it's allowed to be renamed at this point. If we can't rename it, I'll revert the change to this file and use a suppression instead.

**Comment by sharwell** [comment]
@harishsk This is now ready for review. Please make specific note of the indicated public API change.

File: src/Microsoft.ML.Sweeper/AsyncSweeper.cs (hunk 1 above)
Comment (eerhardt) [reply src/Microsoft.ML.Sweeper/AsyncSweeper.cs]: `Microsoft.ML.Sweeper` hasn't shipped in any nuget package, so I think we are safe to rename this.

**Comment by harishsk** [comment]
The changes have been approved. Can you please rebase and merge?

**Pull request merged**